includes/payment-methods/class-wc-stripe-upe-payment-method.php

```diff
@@ -0,0 +1,122 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Abstract UPE Payment Method class
+ *
+ * Handles general functionality for UPE payment methods
+ */
+
+
+/**
+ * Extendable abstract class for payment methods.
+ */
+abstract class WC_Stripe_UPE_Payment_Method {
+
+	/**
+	 * Stripe key name
+	 *
+	 * @var string
+	 */
+	protected $stripe_id;
+
+	/**
+	 * Display title
+	 *
+	 * @var string
+	 */
+	protected $title;
+
+	/**
+	 * Can payment method be saved or reused?
+	 *
+	 * @var bool
+	 */
+	protected $is_reusable;
+
+	/**
+	 * Instance of WC Payments Token Service to save payment method
+	 *
+	 * @var WC_Payments_Token_Service
+	 */
+	protected $token_service;
+
+	/**
+	 * Create instance of payment method
+	 *
+	 * @param WC_Payments_Token_Service $token_service Instance of WC_Payments_Token_Service.
+	 */
+	public function __construct( $token_service ) {
+		//      $this->token_service = $token_service;
```

Comment:
Is this supposed to be commented out?

Reply:
Right now, yeah. The token service will be ported in a separate PR.